resources/templates/md/add-basic-search.md

```diff
@@ -0,0 +1,118 @@
+
+## Prerequisites
+Before we start Node.js and some modules must be installed on your system:
+
+1. Install [`Node.js`](https://nodejs.org/en/download/)
+2. Install lunr: `npm install lunr`
+3. Install cheerio: `npm install cheerio`
+
+## Copy build-index.js
+As the filename says, this script will create the index file, which will be used to search your page.
+
+> *NOTE*: Everytime when a new post/page is published, the index should be recreated. Therefore, the best option would be to add this to your deployment routine.
+
+Download: [build-index.js](/lunr/build-index.js)
+
+### Most notable lines
+
+The following lines might require attention:
+
+Line 8: The output folder of the html files. In this example only **posts-output** will be indexed (no tags, no pages).
+```js
+const HTML_FOLDER = "public/posts-output";
+```
+
+Line 11: Files to exclude from the index
+```js
+const EXCLUDE_FILES = ["search.html"];
+```
+
+Line 13: The output path and filename for the index. This depends on your actual theme.
+```js
+const OUTPUT_INDEX = "themes/nucleus/js/index/lunr_index.js";
+```
+
+Line 64: This adds "posts-output" to the link, which might be removed depending on you page
+```js
+        "link": "posts-output/"+file,
+```
+
+## Create index file
+The index file should be created with any new post or page which is published using the following command:
+```bash
+node ./build-index.js
+```
+## Download lunrclient.js
+Download and place the file lunrclient.js into the js folder of your theme: [lunrclient.js](/lunr/lunrclient.js)
+
+This script will create the search rusult page.
+
+## Changes to HTML and CSS
+Changes to the actual html/css highly depend on your theme. Therefor only a basic snippets of a possible change of the **base.html** are shown here.
+
+In the `head` tag the Javascripts must be loaded. First lunr itself, then the indexfile and then the client file:
+```html
+    <script src="https://unpkg.com/lunr/lunr.js"></script>
+    <script src="/js/index/lunr_index.js"></script>
+    <script src="/js/lunrclient.js"></script>
+```
+
+The search form can look like this:
+```html
+    <!--- Search from -->
+    <form id="lunrSearchForm" name="lunrSearchForm" action="/">
+        <input class="search-input" name="q" placeholder="Enter search term" type="text"> 
+        <input class="button nopadding" type="submit" value="Search">
+    </form>
+    <div class="loading" id="loading-div">
+        <i class="fa fa-spinner fa-pulse fa-3x fa-fw"></i>
+        <span class="sr-only">Loading...</span>
+    </div>
+```
+
+Last but not least the search result is rendered in the main page (right side, nucleus example theme was used here.)
+The script lunrclient.js will replace the elements "resultCount" and "searchResults"
+```html
+<div id="right" class="clearfix">
+  <div class="resultCount" id="resultCount"></div>
+  <hr />
+  <div id="searchResults">
+    {% block content %}
```

Comment:
why the 

```
{% block content %}
    {% endblock %}
```
?

Reply:
well, that was long time ago to be honest. I quickly tested it and if I remove it, the pages are empty. 
thats actually the blog content, which I framed into the div for the searchResult. 
If I recall correctly, that was the only good looking result